plugins/fireeye_hx/icon_fireeye_hx/util/api.py

```diff
@@ -0,0 +1,62 @@
+from komand.exceptions import PluginException
+import requests
+from requests import Session
+from requests.auth import HTTPBasicAuth
+from urllib.parse import urlsplit
+import json
+from icon_fireeye_hx.util import endpoint
+
+
+class FireEyeAPI:
+    def __init__(self, url, username, password, ssl_verify, logger):
+        self.url = f"{self.split_url(url)}/hx/api/v3"
+        self.ssl_verify = ssl_verify
+        self.logger = logger
+        self.session = Session()
+        self.session.auth = HTTPBasicAuth(username=username, password=password)
+
+    def call_api(
+        self, path: str, method: str = "GET", params: dict = None, json_data: dict = None, data: dict = None
+    ) -> requests.Response:
+        try:
+            response = self.session.request(
+                method=method.upper(),
+                url=f"{self.url}/{path}",
+                json=json_data,
+                data=data,
+                params=params,
+                verify=self.ssl_verify,
+            )
+            if response.status_code == 401:
+                raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD, data=response.text)
+            if response.status_code == 403:
+                raise PluginException(preset=PluginException.Preset.API_KEY, data=response.text)
```

Comment:
```suggestion
            if response.status_code in [401, 403]:
                raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD, data=response.text)
```
Combining these statements as there is no API key used in this plugin, just username & password